Segment Tree/LazyPropagation/LazyPropagation.playground/Contents.swift

```diff
@@ -0,0 +1,129 @@
+public class LazySegmentTree {
+    
+    private var value: Int
+    
+    private var leftBound: Int
+    
+    private var rightBound: Int
+    
+    private var leftChild: LazySegmentTree?
+    
+    private var rightChild: LazySegmentTree?
+    
+    // Interval Update Lazy Element
+    private var lazyValue: Int
+    
+    // MARK: - Push Up Operation
+    // Description: pushUp() - update items to the top
+    private func pushUp(lson: LazySegmentTree, rson: LazySegmentTree) {
+        self.value = lson.value + rson.value
+    }
+    
+    // MARK: - Push Down Operation
+    // Description: pushDown() - update items to the bottom
+    private func pushDown(round: Int, lson: LazySegmentTree, rson: LazySegmentTree) {
+        guard lazyValue != 0 else { return }
+        lson.lazyValue += lazyValue
+        rson.lazyValue += lazyValue
+        lson.value += lazyValue * (round - (round >> 1))
+        rson.value += lazyValue * (round >> 1)
+        lazyValue = 0
+    }
+    
+    public init(array: [Int], leftBound: Int, rightBound: Int) {
+        self.leftBound = leftBound
+        self.rightBound = rightBound
+        self.value = 0
+        self.lazyValue = 0
+        
+        guard leftBound != rightBound else {
+            value = array[leftBound]
+            return
+        }
+        
+        let middle = leftBound + (rightBound - leftBound) / 2
+        leftChild = LazySegmentTree(array: array, leftBound: leftBound, rightBound: middle)
+        rightChild = LazySegmentTree(array: array, leftBound: middle + 1, rightBound: rightBound)
+        if let leftChild = leftChild, let rightChild = rightChild {
+            pushUp(lson: leftChild, rson: rightChild)
+        }
+    }
+    
+    public convenience init(array: [Int]) {
+        self.init(array: array, leftBound: 0, rightBound: array.count - 1)
+    }
+    
+    public func query(leftBound: Int, rightBound: Int) -> Int {
+        if leftBound <= self.leftBound && self.rightBound <= rightBound {
+            return value
+        }
+        guard let leftChild  = leftChild  else { fatalError("leftChild should not be nil") }
+        guard let rightChild = rightChild else { fatalError("rightChild should not be nil") }
+        
+        pushDown(round: self.rightBound - self.leftBound + 1, lson: leftChild, rson: rightChild)
+        
+        let middle = self.leftBound + (self.rightBound - self.leftBound) / 2
+        var result: Int = 0
```

Comment:
since it has already got a value. Swift will get the correct type. No need to specify it.